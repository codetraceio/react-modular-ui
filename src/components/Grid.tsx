import React = require("react");

import componentService, { IProp } from "../services/componentService";

interface IGridProps {
  columns?: IProp<string | number>;
  space?: IProp<string | number>;
  templateColumns?: string;
}

export default class Grid extends React.PureComponent<IGridProps, {}> {
  getModifiers() {
    return {
      columns: this.props.columns,
      space: this.props.space,
    };
  }

  getStyle() {
    return {
      gridTemplateColumns: this.props.templateColumns,
    };
  }

  render() {
    return (
      <div
        className={componentService.getBlockName("grid", this.getModifiers())}
        style={this.getStyle()}
      >
        {this.props.children}
      </div>
    );
  }
}
