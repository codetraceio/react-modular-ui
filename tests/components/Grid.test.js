import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Grid from '../../src/components/Grid';

describe('Grid', () => {
  it('should render Grid', () => {
    const tree = renderer.create(
      <Grid
        columns="3"
        space="16"
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
