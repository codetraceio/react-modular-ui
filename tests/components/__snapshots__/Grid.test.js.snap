// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render Grid 1`] = `
<div
  className="c-grid -columns-3 -space-16"
  style={
    Object {
      "gridTemplateColumns": undefined,
    }
  }
/>
`;
